frontend/src/views/CellGuide/components/common/OntologyDagView/index.tsx

```diff
@@ -209,69 +258,42 @@ export default function OntologyDagView({
-          initialTreeState?.notShownWhenExpandedNodes ?? {};
-        // If this node is a key in `notShownWhenExpandedNodes`, then it is a parent to nodes
-        // that should be collapsed. Therefore, set `appendDummy` to `true`.
-        // The term "dummy" here is used to describe the collapsed node containing hidden terms.
-        // The text label of a "dummy" is the number of hidden terms, e.g. 52 cell types.
-        const appendDummy = d.id in notShownWhenExpandedNodes;
-
-        // `showAllChildren` is a flag that is set to `true` when the user clicks on a collapsed node.
-        // It indicates that all of the children of the node should be shown.
-        for (const child of d?.children ?? []) {
+          initialTreeState.notShownWhenExpandedNodes;
+        /**
+         * If a node is a key in `notShownWhenExpandedNodes`, then it is a parent to nodes
+         * that should be collapsed. The text label of this "dummy" node is the number of hidden terms,
+         * e.g. 52 cell types.
+         * `showAllChildren` is a flag that is set to `true` when the user clicks on a collapsed node.
+         * It indicates that all of the children of the node should be shown.
+         */
+
+        for (const child of d.children) {
           if (
             d.showAllChildren ||
-            !appendDummy ||
-            // This is a child that should be shown because it is in a path to the target node.
-            (appendDummy && !notShownWhenExpandedNodes[d.id].includes(child.id))
+            !notShownWhenExpandedNodes[d.id]?.includes(child.id)
           ) {
             newChildren.push(child);
           }
         }
 
-        if (appendDummy && !d.showAllChildren) {
-          const numHiddenChildren =
-            (d.children?.length ?? 0) - newChildren.length;
+        const numHiddenChildren = d.children.length - newChildren.length;
+        if (numHiddenChildren > 0) {
           newChildren.push({
             id: `dummy-child-${d.id}`,
             name: `${numHiddenChildren} cell types`,
             n_cells: 0,
             n_cells_rollup: 0,
             isExpanded: false,
           });
-        } else if (d.showAllChildren) {
-          // Sorting such that the first child in the `newChildren` array is the
-          // same as the first child in the original array (`d.children`) seems to
-          // ensure that the position of the visible node is preserved when its dummy
-          // sibling node is expanded. It makes for smoother animations.
-          if (d.children) {
-            const firstId = d.children[0].id;
-            newChildren.sort((a, b) => {
-              if (a.id === firstId) return -1;
-              if (b.id === firstId) return 1;
-              return a.id.localeCompare(b.id);
-            });
-          }
         }
+        return newChildren;
       }
-      return newChildren.length ? newChildren : null;
     });
   }, [treeData, triggerRender, initialTreeState]);
```

Comment:
looks like we can remove `triggerRender`?

<img width="1045" alt="Screenshot 2023-08-14 at 8 11 32 AM" src="https://github.com/chanzuckerberg/single-cell-data-portal/assets/6309723/5c1b6df1-4708-4b7b-a682-fae48c57c244">